### ISSUE TITLE
/works endpoint is broken

### ISSUE THREAD

**Issue by alexwlchan** [wellcometrust/platform-api#368]
```console
$ curl 'https://api.wellcomecollection.org/catalogue/v0/works'
{"message":"[instance-0000000000][172.17.0.65:19796][indices:data/read/search]"}
```

Experimenting with local builds suggests that it's the sort on canonicalId that's playing up – remove that, and all's well.

**Comment by alexwlchan** [comment]
Trying a sort directly against the ES cluster:

```json
{
  "error": {
    "root_cause": [
      {
        "type": "illegal_argument_exception",
        "reason": "Fielddata is disabled on text fields by default. Set fielddata=true on [canonicalId] in order to load fielddata in memory by uninverting the inverted index. Note that this can however use significant memory. Alternatively use a keyword field instead."
      }
    ],
    "type": "search_phase_execution_exception",
    "reason": "all shards failed",
    "phase": "query",
    "grouped": true,
    "failed_shards": [
      {
        "shard": 0,
        "index": "works",
        "node": "DAXQOUGAQqqSueZs9zvcLg",
        "reason": {
          "type": "illegal_argument_exception",
          "reason": "Fielddata is disabled on text fields by default. Set fielddata=true on [canonicalId] in order to load fielddata in memory by uninverting the inverted index. Note that this can however use significant memory. Alternatively use a keyword field instead."
        }
      }
    ]
  },
  "status": 400
}
```

**Comment by alexwlchan** [comment]
I’ve checked against a test index – we get a `keyword` field if we create the index from scratch. I think the most likely sequence of events is this:

* We deleted the index.
* An ingestor was already running, and tried to index a document.
* Elasticsearch "helpfully" created the index, and created a dynamic mapping based on the document it saw.
* When the ingestor tries to PUT the mapping, Elasticsearch adds it to the existing mapping, forcing the keyword field down a level.

This should be caught by having a `strict` mapping – but that doesn't exist until we've created the index!

There's an option in the ElasticCloud settings to disable automatic index creation. I’ve turned that off, then I’ll delete the existing index and restart an ingestor. I don’t think there’s a way to automate having that setting off, alas.
